x/poolmanager/router.go

```diff
@@ -559,9 +560,11 @@ func (k Keeper) ListPoolsByDenom(
 
 		var poolsByDenom []types.PoolI
 		for _, pool := range currentModulePools {
+			// If the pool is incorrectly implemented and we can't get the PoolDenoms
+			// skip the pool.
 			poolDenoms, err := poolModule.GetPoolDenoms(ctx, pool.GetId())
 			if err != nil {
-				return nil, err
+				continue
```

Comment:
I'm a bit worried about simply skipping any error. What if we instrumented a CW-pool specific error at a lower level of abstraction and checked for it here?

Reply:
I don't understand the suggestion, or an alternate to skipping here

Reply:
A correct pool shouldn't error on this function. If it does, it's invalid code and thus cannot be handled.

I think we should remove errors from this api and make cw pools just return a nil denlm list if they have an internal error

Reply:
The original issue was created due some CW pools not being implemented correctly and making the query fail as a whole.

Since we conclude that it is acceptable to skip such errors, I'm suggesting that we should consider to limit what we skip by:
```go
if err != nil && errors.Is(err, SpecificCosmWasmPoolError{}) {
    continue
} else if err != nil {
   return err
}
```

if something else breaks unexpectedly downstream, we might not catch it timely by skipping all errors silently

Reply:
But I don't see what error could possibly occur that would have a correct behavior other than be skipped here.